app/Http/Requests/StoreSong.php

```diff
@@ -24,7 +24,13 @@ public function authorize()
     public function rules()
     {
         return [
-            //
+            'name' => 'required|string|max:255',
+            'alt_name' => 'string|max:255',
+            'flac_file' => 'required|file|max:2000000',
+            'track_number' => 'integer|max:99',
+            'preview_start' => 'between:0.000,9999.999',
```

Comment:
track_number and preview_start be required, too.

Reply:
These have default values not making them required though ?

```php
$table->unsignedTinyInteger('track_number')->default(1);
$table->unsignedDecimal('preview_start', 7, 3)->default(0.000);
```

Reply:
Excellent; we're all set, then.